### PR TITLE
Adding xraylarch recipe

### PR DESCRIPTION
**Pull request by kbeeperez**
implements: https://github.com/nsls-ii-forge/xraylarch-feedstock

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by kbeeperez** [comment]
Attn @mrakitin

**Review by mrakitin — APPROVED** [review]
Thanks, @kbeeperez, looks good to me. It's worth mentioning that there was a discussion with the author that the binaries can be preserved (see https://github.com/nsls-ii-forge/xraylarch-feedstock/issues/10 for the link). We will address it in the feedstock for the new version.

I agree to maintain this package.

**Comment by mrakitin** [comment]
@conda-forge/staged-recipes, this is ready to go!

**Comment by kbeeperez** [comment]
@conda-forge/staged-recipes This is good to go.

File: recipes/xraylarch/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  url: https://github.com/xraypy/{{ name }}/archive/{{ version }}.tar.gz
+  sha256: f64eee63bca13a67d313a06c9b437502ea59a5ca961075756ebf453e9dc1da81
+  patches:
+    - 0001-MNT-remove-the-desktop-shortcut-logic.patch
+    - 0002-MNT-remove-the-feff-wrappers-and-binaries.patch
+
+build:
+  number: 0
+  script: |
+    # We do not have a license to distribute built FEFF binaries.
```
Comment (ocefpaf) [inline recipes/xraylarch/meta.yaml:18]: LGTM but I'm not an expert on the topic. @isuruf do you think that removing the binaries "covers" it with regarding to license to re-distribute/package this?

Comment (mrakitin) [reply recipes/xraylarch/meta.yaml:18]: We can try to remove the patch, I think it's OK now without it.

File: recipes/xraylarch/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    - 0001-MNT-remove-the-desktop-shortcut-logic.patch
+
+build:
+  number: 0
+  script: |
+    # We do not have a license to distribute built FEFF binaries.
+    # We may have one to deploy it to the floor, but definitely not to make
+    # them public, so remove binaries.
+    rm -rf larch/bin  # [not win]
+    rmdir /q /s larch/bin  # [win]
```
Comment (mrakitin) [inline recipes/xraylarch/meta.yaml:21]: We can now remove these lines above.

**Review by mrakitin — COMMENTED** [review]
I have a single suggestion based on the recently removed patch.

**Comment by kbeeperez** [comment]
the build fails with an overlinking error: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=358288&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1852

We will be keeping the patches in.

File: recipes/xraylarch/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    {{ PYTHON }} -m pip install . --no-deps -vv
+  skip: true  # [py<36]
+
+requirements:
+  host:rm 00
```
Comment (mrakitin) [inline recipes/xraylarch/meta.yaml:27]:
```suggestion
  host:
```

**Comment by kbeeperez** [comment]
@conda-forge/staged-recipes ready to merge.

**Comment by kbeeperez** [comment]
@conda-forge/staged-recipes

File: recipes/xraylarch/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    - scikit-learn
+    - scipy >=1.2
+    - silx
+    - six >=1.10
+    - sqlalchemy >=0.9
+    - termcolor
+    - uncertainties
+    - xraydb >=4.3
+
+test:
```
Comment (xhochy) [inline recipes/xraylarch/meta.yaml:54]:
```suggestion
test:
  requires:
    - pip
  commands:
    - pip check
```

**Comment by mrakitin** [comment]
Seems like it needs another package for Windows:
```py
Traceback (most recent call last):
  File "C:\Miniconda\conda-bld\pyshortcuts_1627932418092\test_tmp\run_test.py", line 2, in <module>
    import pyshortcuts
  File "C:\Miniconda\conda-bld\pyshortcuts_1627932418092\_test_env\lib\site-packages\pyshortcuts\__init__.py", line 23, in <module>
    from .windows import (scut_ext, ico_ext, make_shortcut,
  File "C:\Miniconda\conda-bld\pyshortcuts_1627932418092\_test_env\lib\site-packages\pyshortcuts\windows.py", line 11, in <module>
    import win32com.client
ModuleNotFoundError: No module named 'win32com'
```

**Review by mrakitin — COMMENTED** [review]
@kbeeperez, the `pyshortcuts/meta.yaml` recipe should be made arch-dependent in my opinion, and add the `pywin32` as a dependency with the selector `# [win]` in the run dependencies.

File: recipes/pyshortcuts/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/pyshortcuts-{{ version }}.tar.gz
+  sha256: 1e66467a0b7c15f42e2d616abeb2d80aa81d9e0ccca50991dc79bdb99f39d8c1
+
+build:
+  number: 0
+  noarch: python
```
Comment (mrakitin) [inline recipes/pyshortcuts/meta.yaml:15]:
```suggestion
```

File: recipes/pyshortcuts/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  entry_points:
+    - pyshortcut = pyshortcuts:shortcut_cli
+  script: {{ PYTHON }} -m pip install . -vv
```
Comment (mrakitin) [inline recipes/pyshortcuts/meta.yaml:18]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv --no-deps
```

File: recipes/pyshortcuts/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+requirements:
+  host:
+    - pip
+    - python >=3.6
+  run:
+    - python >=3.6
```
Comment (mrakitin) [inline recipes/pyshortcuts/meta.yaml:25]:
```suggestion
    - python
    - pywin32  # [win]
```

File: recipes/pyshortcuts/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+  number: 0
+  noarch: python
+  entry_points:
+    - pyshortcut = pyshortcuts:shortcut_cli
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - pip
+    - python >=3.6
```
Comment (mrakitin) [inline recipes/pyshortcuts/meta.yaml:23]:
```suggestion
    - python
```

File: recipes/pyshortcuts/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,46 @@
+{% set name = "pyshortcuts" %}
+{% set version = "1.8.0" %}
+
+
```
Comment (mrakitin) [inline recipes/pyshortcuts/meta.yaml:4]:
```suggestion
```

File: recipes/xraylarch/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+    - lmfit >=1.0
+    - matplotlib >=3.0
+    - numpy >=1.7
+    - peakutils >=1.0.0
+    - pillow >=3.4
+    - psutil
+    - pyFAI
+    - pyshortcuts >=1.7.2
+    - pyyaml
+    - pywin32  # [win]
```
Comment (mrakitin) [inline recipes/xraylarch/meta.yaml:44]: The `pyshortcuts` package should bring that dependency.
```suggestion
```

**Review by mrakitin — CHANGES_REQUESTED** [review]
Here are my suggestions to fix it.

**Comment by mrakitin** [comment]
🎉

**Comment by mrakitin** [comment]
@xhochy, this is ready to go. Thanks for your suggestions about `pip check`, which definitely uncovered some hidden issues.

**Comment by kbeeperez** [comment]
@conda-forge/staged-recipes

**Comment by kbeeperez** [comment]
@conda-forge/staged-recipes

**Comment by mrakitin** [comment]
Thanks a lot, @xhochy!

**Pull request merged**